packages/cardano-services/test/util/TypeormProvider.test.ts

```diff
@@ -0,0 +1,43 @@
+import { BehaviorSubject, of } from 'rxjs';
+import { TypeormProvider } from '../../src/util';
+import { getEntities } from '../../src';
+import { logger } from '@cardano-sdk/util-dev';
+import { projectorConnectionConfig } from '../util';
+
+class TestProvider extends TypeormProvider {}
+
+jest.mock('../../src/util/createTypeormDataSource', () => ({
+  createTypeormDataSource: jest.fn().mockImplementation(() => {
+    let called = false;
+
+    return of({
+      query: () => {
+        if (called) return Promise.reject(0);
+
+        called = true;
+        return Promise.resolve(0);
+      }
+    });
+  })
+}));
+
+describe('TypeormProvider', () => {
+  const connectionConfig$ = new BehaviorSubject(projectorConnectionConfig);
```

Comment:
Would be nice to make it clear that this doesn't matter for these tests
```suggestion
  const notUsedConnectionConfig = {} as ConnectionConfig;
  const connectionConfig$ = of(notUsedConnectionConfig)
```

Reply:
d8d401bd29 should solve this